Mathlib/Analysis/SpecialFunctions/Arcosh.lean

```diff
@@ -0,0 +1,256 @@
+/-
+Copyright (c) 2025 Yuval Filmus. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yuval Filmus
+-/
+module
+
+public import Mathlib.Analysis.SpecialFunctions.Trigonometric.Deriv
+public import Mathlib.Analysis.SpecialFunctions.Log.Basic
+
+/-!
+# Inverse of the cosh function
+
+In this file we define an inverse of cosh as a function from [0, ∞) to [1, ∞).
+
+## Main definitions
+
+- `Real.arcosh`: An inverse function of `Real.cosh` as a function from [0, ∞) to [1, ∞).
+
+- `Real.cosh'`, `Real.arcosh'`: Versions of `Real.cosh` and `Real.arcosh` restricted
+  to the domains [0, ∞) and [1, ∞), respectively.
+
+- `Real.cosh'Equiv`, `Real.cosh'OrderIso`, `Real.cosh'Homeomorph`: `Real.cosh'` as an `Equiv`,
+  `OrderIso`, and `Homeomorph`, respectively.
+
+## Main Results
+
+- `Real.cosh_arcosh`, `Real.arcosh_cosh`: cosh and arcosh are inverse in the appropriate domains.
+
+## Tags
+
+arcosh, arccosh, argcosh, acosh
+-/
+
+@[expose] public section
+
+
+noncomputable section
+
+open Function Filter Set NNReal
+
+open scoped Topology
+
+namespace Real
+
+variable {x y : ℝ}
+
+/-- `arcosh` is defined using a logarithm, `arcosh x = log (x + √(x ^ 2 - 1))`. -/
+@[pp_nodot]
+def arcosh (x : ℝ) :=
+  log (x + √(x ^ 2 - 1))
+
+theorem exp_arcosh (x : ℝ) (hx : 1 ≤ x) : exp (arcosh x) = x + √(x ^ 2 - 1) := by
+  apply exp_log
+  positivity
+
+@[simp]
+theorem arcosh_zero : arcosh 1 = 0 := by simp [arcosh]
+
+lemma x_add_sqrt_x_sq_sub_one_inv (x : ℝ) (hx : 1 ≤ x) :
+  (x + √(x ^ 2 - 1))⁻¹ = x - √(x ^ 2 - 1) := by
+    apply inv_eq_of_mul_eq_one_right
+    rw [← pow_two_sub_pow_two, sq_sqrt (sub_nonneg_of_le (one_le_pow₀ hx)), sub_sub_cancel]
```

Comment:
1. Naming: we don't refer to variables by name in declarations (because they could change!). They are omitted, but if you want to say that a certain one is the same, you can use `self`, like so.
2. Style: indentation
```suggestion
lemma add_sqrt_self_sq_sub_one_inv (x : ℝ) (hx : 1 ≤ x) :
    (x + √(x ^ 2 - 1))⁻¹ = x - √(x ^ 2 - 1) := by
  apply inv_eq_of_mul_eq_one_right
  rw [← pow_two_sub_pow_two, sq_sqrt (sub_nonneg_of_le (one_le_pow₀ hx)), sub_sub_cancel]
```